src/NUnitFramework/tests/Internal/RuntimeFrameworkTests.cs

```diff
@@ -26,17 +26,17 @@ public void CanGetCurrentFramework()
             Assert.That(framework.ClrVersion, Is.EqualTo(Environment.Version), "#2");
         }
 
-#if NET45
+#if NETFRAMEWORK
         [Test]
         public void TargetFrameworkIsSetCorrectly()
         {
             // We use reflection so it will compile and pass on Mono,
             // including older versions that do not have the property.
-            var prop = typeof(AppDomainSetup).GetProperty("FrameworkName");
+            var prop = typeof(AppDomainSetup).GetProperty("TargetFrameworkName");
```

Comment:
I was a little confused by this, it looks like this prop is actually named `TargetFrameworkName` in the .NET Framework, so I update this test accordingly

Reply:
So how did this test ever pass?
I see the test uses `Assume.That(prop, Is.Not.Null);` which marks the test as inconclusive instead of failed.
Never heard of that class. We would use `Assert.That(prop, Is.Not.Null);` which would cause the test to fail.

Looking at the last test run on master: `Test Count: 5823, Passed: 5641, Failed: 0, Warnings: 0, Inconclusive: 120, Skipped: 62` There are 120 inconclusive tests. How many of these are hidden failures because of failing assumptions?
Not sure what the idea is/was behind `Assume`.

Reply:
This test is now failing in the Linux build: 
```
Failed : NUnit.Framework.Internal.RuntimeFrameworkTests.TargetFrameworkIsSetCorrectly
  Expected: String starting with ".NETFramework,Version=v4."
  But was:  null
```

This test should never run on Linux as Linux doesn't support .NET Framework.

Reply:
Thanks for checking this out :D 
I'll have to take a closer look, but prior to my change (on my local windows) the behaviour I saw was:
- It was wrapped in a `#if NET45`
- It was "Inconclusive" because the prop lookup was returning null on Windows
Now:
- It is wrapped in a `#if NETFRAMEWORK`
- It is passing on Windows
- It is (apparently) running + failing on linux
Will take a bit of digging. I'm very curious about this myself too

Reply:
Alrighty, after some digging this appears to be a mono-specific issue like the comment indicates. Tricky thing is, mono can be installed on Windows in additional to *nix (though our CI doesn't test mono on Windows). I suspect that this test had been "inconclusive" for a while though I assume it must've worked at some point in the past. I can't find a previous version of the .NET Framework which has a `AppDomainSetup.FrameworkName` property, however.

At any rate, either of the two options seem to work:

- Use `[Platform("Win")]`
  - This works for us since the netfx tests only mono on linux/mac
- Use the `Assume.That(Is.Not.Null)` approach on the _return value_ of the prop invocation rather than the prop returned via reflection
  - This allows us to work-around the runtime differences without getting into OS differences, but has the potential to silently turn "inconclusive" again in the future

I've left each of these approaches as separate commits in the history (the last 2 commits in this PR). No strong preference from me on either, other than that I slightly prefer the second because it allows us to test the `RuntimeFramework` class without relying on an attribute which _also_ relies on the class under test.

Reply:
Wait, this doesn't actually test nunit code from what I can tell. Oops

Reply:
Alrighty, I have removed these tests completely. My impression is that they were added to test netfx-specific functionality introduced in NET45 (which we now no longer support). I'd appreciate hearing any feedback the team may have about this.